File: src/main/java/org/teamvoided/transition/config/TransitionConfig.java
package org.teamvoided.transition.config;

import me.fzzyhmstrs.fzzy_config.config.Config;
import net.minecraft.resources.ResourceLocation;
import org.teamvoided.transition.Transition;

import java.util.List;

public class TransitionConfig extends Config {

    public MappingModes mode = MappingModes.ON_LOAD;
    public List<String> directoryBlackList = List.of("datapacks", "mods", "config");

    public TransitionConfig() {
        super(ResourceLocation.fromNamespaceAndPath(Transition.MODID, "main"));
    }
}

File: src/main/kotlin/org/teamvoided/transition/config/TransitionConfig.kt
package org.teamvoided.transition.config

import me.fzzyhmstrs.fzzy_config.config.Config
import org.teamvoided.transition.Transition.id

class TransitionConfig : Config(id("main")) {
    @JvmField
    var mode = MappingModes.ON_LOAD
    var directoryBlackList: MutableList<String> = mutableListOf("datapacks", "mods", "config")
}
